Datahub.Web/Pages/Search.cshtml.cs

```diff
@@ -1,49 +1,71 @@
-using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
-using Datahub.Web.Data;
 using Datahub.Web.Models;
-using Datahub.Web.Pages.Helpers;
 using Microsoft.AspNetCore.Hosting;
 using Microsoft.AspNetCore.Mvc.RazorPages;
+using Datahub.Web.Elasticsearch;
+using Nest;
+using Microsoft.AspNetCore.Mvc;
 
 namespace Datahub.Web.Pages
 {
     public class SearchModel : PageModel
     {
-        readonly IHostingEnvironment env;
-        public SearchModel(IHostingEnvironment env)
-        {
-            this.env = env;
-        }
+        public const string DefaultIndex = "main";
+        public const string DefaultSite = "datahub";
+        public const int DefaultSize = 10;
+        public const int DefaultStart = 0;
 
-        public List<SearchResultModel> Results { get; set; }
-        public List<KeywordModel> Keywords { get; set; }
+        private readonly IHostingEnvironment _env;
+        private readonly ElasticClient _client;
 
+        public ISearchResponse<SearchResult> Results { get; set; }
 
-        public async Task OnGetAsync(string q, string[] k)
-        {
-            // populate Keywords to show which are filtered on
-            this.Keywords = ParseKeywords(k);
+        [BindProperty(Name = "q", SupportsGet = true)]
+        public string QueryString{ get; set; }
+        [BindProperty(Name = "p", SupportsGet = true)]
+        public int CurrentPage { get; set; }
+        [BindProperty(Name = "s", SupportsGet = true)]
+        public int CurrentPageSize { get; set; }
 
-            // populate Results
-            var assets = await JsonLoader.LoadAssets(this.env.ContentRootPath);
+        public List<Keyword> Keywords { get; set; }
 
-            var query = ApplyQuery(assets);
+        public SearchModel(IHostingEnvironment env, IElasticsearchService elasticsearchService)
+        {
+            _env = env;
+            _client = elasticsearchService.Client();
```

Comment:
It's s bit weird to store a reference to just the client here. The convention when doing DI is that you store the reference to the thing injected.